django/db/models/fields/related_descriptors.py

```diff
@@ -0,0 +1,1039 @@
+"""
+Accessors for related objects.
+
+When a field defines a relation between two models, each model class provides
+an attribute to access related instances of the other model class (unless the
+reverse accessor has been disabled with related_name='+').
+
+Accessors are implemented as descriptors in order to customize access and
+assignment. This module defines the descriptor classes.
+
+Forward accessors follow foreign keys. Reverse accessors trace them back. For
+example, with the following models::
+
+    class Parent(Model):
+        pass
+
+    class Child(Model):
+        parent = ForeignKey(Parent, related_name='children')
+
+ ``child.parent`` is a forward many-to-one relation. ``parent.children`` is a
+reverse many-to-one relation.
+
+There are three types of relations (many-to-one, one-to-one and many-to-many)
+and two directions (forward and reverse) for a total of six combinations.
+
+1. Related instance on the forward side of a many-to-one or one-to-one
+   relation: ``ForwardManyToOneDescriptor``.
+
+   Unicity of foreign key values is irrelevant to accessing the related
```

Comment:
Unicity -> Uniqueness? or "Whether or not a foreign key has unique=True is..." (I'm a bit concerned for non-native speakers when I have to look up a word).